### DISCUSSION TITLE
Plugin location - Debian 10

### DISCUSSION THREAD

**Discussion by emresaracoglu** [streamlink/streamlink#4475] [Q&A] [answered]
Hello,

Where is the Streamlink Plugin folder in Debian 10 operating system? I looked through the documentation but couldn't find it.

```
web2@serve:~$ cd /usr/share/streamlink
web2@serve:/usr/share/streamlink$ ls
streamlink  streamlink_cli
web2@serve:/usr/share/streamlink$ ^C
web2@serve:/usr/share/streamlink$ cd streamlink_cli/
web2@serve:/usr/share/streamlink/streamlink_cli$ ls
argparser.py  compat.py  console.py  constants.py  __init__.py  __main__.py  main.py  output.py  __pycache__  utils
web2@serve:/usr/share/streamlink/streamlink_cli$
```

Thank you!

**Comment by WolfganP** [comment]
As per the docs https://streamlink.github.io/cli.html#plugin-specific-configuration-file plugins should be sideloaded at ${XDG_CONFIG_HOME:-${HOME}/.config}/streamlink/config.pluginname
If you want to try a custom plugin or new version of an existing one, sideloading is the way...

**Reply by emresaracoglu** [reply]
Hello, @WolfganP Thank you for your answer, `${XDG_CONFIG_HOME:-${HOME}` how do I find this location?

**Reply by gravyboat** [reply]
We don't have the bandwidth to walk people through how to use Linux based operating systems. Pleas try Googling.

**Comment by bastimeyer** [comment, marked as answer]
**Before asking questions, it is expected to at least do a bit of research yourself.** This is not a commercial product, so anyone answering questions is doing it in their free time out of good will and nothing else, and asking low effort questions just steals everyone's time and is unnecessary.

----

In order to find the plugin locations, you could've easily checked the package contents via
```sh
$ dpkg-query -L python3-streamlink | grep plugins
/usr/lib/python3/dist-packages/streamlink/plugins
...
```

As @WolfganP already mentioned, if you want to modify plugins or add custom ones, you are supposed to **sideload** them, and not change any package contents, as this not only requires root access, but also overwrites stuff on the next update.

https://streamlink.github.io/latest/cli/plugin-sideloading.html

The location for sideloading plugins is `${XDG_DATA_HOME:-${HOME}/.local/share}/streamlink/plugins`. This is shell syntax which tells that if `XDG_DATA_HOME` is unset, the default value of `${HOME}/.local/share` is used instead. This means that the path for sideloading plugins is either `${XDG_DATA_HOME}/streamlink/plugins` if `XDG_DATA_HOME` is set, or `${HOME}/.local/share/streamlink/plugins` if it is unset. This is part of the [XDG base directory specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) on Linux, which defines standard paths for applications and developers.

**Reply by emresaracoglu** [reply]
**Thank you again** @bastimeyer
